pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -496,6 +506,23 @@ func managePod(ctx context.Context, configMapsGetter corev1client.ConfigMapsGett
 	return resourceapply.ApplyConfigMap(configMapsGetter, recorder, configMap)
 }
 
+func readKubeControllerManagerArgs(config map[string]interface{}) string {
+	extendedArguments, ok := config["extendedArguments"]
+	if !ok || extendedArguments == nil {
+		return ""
+	}
+	args := []string{}
+	for key, value := range extendedArguments.(map[string]interface{}) {
+		for _, arrayValue := range value.([]interface{}) {
+			args = append(args, fmt.Sprintf("--%s=%s", key, arrayValue.(string)))
+		}
+	}
+	// make sure to sort the arguments, otherwise we might get mismatch
+	// when comparing revisions leading to new ones being created, unnecessarily
+	sort.Strings(args)
```

Comment:
Have you checked the yaml parser version? I thought there was a bug that changed the order but with the correct version you'd not need to sort

Reply:
I don't think the yaml parser has anything to do with it. This sort operation is not that expensive and gives us necessary confidence.

Reply:
For some programs order of arguments matter that's why I'd prefer to avoid the sort and rather figure out why the order changed while decoding. If I change arg order on the resource I'd expect it to rollout.

Reply:
https://godoc.org/gopkg.in/yaml.v2#MapSlice

But since we control the args I can live with the sort as well.